extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/blocking/BlockingServerInterceptor.java

```diff
@@ -87,28 +96,40 @@ public void handle(Promise<Object> f) {
      */
     private class ReplayListener<ReqT> extends ServerCall.Listener<ReqT> {
         private ServerCall.Listener<ReqT> delegate;
-        private final List<Consumer<ServerCall.Listener<ReqT>>> incomingEvents = new LinkedList<>();
+        private final List<ConsumerWithState> incomingEvents = new ArrayList<>();
 
-        synchronized void setDelegate(ServerCall.Listener<ReqT> delegate) {
+        synchronized void setDelegate(ServerCall.Listener<ReqT> delegate,
+                ManagedContext requestContext) {
             this.delegate = delegate;
-            for (Consumer<ServerCall.Listener<ReqT>> event : incomingEvents) {
-                event.accept(delegate);
+            for (ConsumerWithState event : incomingEvents) {
+                requestContext.activate(event.contextState);
```

Comment:
Should we only activate/deactivate once (outside of the loop)?

Reply:
I'll try that

Reply:
fixed, we have a single activate/deactivate for the whole loop now